frontend/src/components/transactions/TransactionDetails.tsx

```diff
@@ -36,9 +38,24 @@ export default class extends React.Component<Props, State> {
         actionArgs = action[actionKind];
       }
       if (actionArgs.hasOwnProperty("deposit")) {
-        this.setState({ deposit: actionArgs.deposit });
+        return actionArgs.deposit;
+      } else {
+        return "0";
       }
     });
+    const IntLengths: Array<number> = deposits.map(dp => dp.length);
+    const max = Math.max(...IntLengths);
+    const bigIntDeposit = deposits
+      .map(dp => new Big(dp))
+      .reduce((current, dp) => dp.plus(current), 0)
+      .c.join("");
+    if (bigIntDeposit.length < max) {
+      const tail = "0".repeat(max - bigIntDeposit.length);
+      const deposit = bigIntDeposit.concat(tail);
+      this.setState({ deposit });
+    } else {
+      this.setState({ deposit: bigIntDeposit });
+    }
```

Comment:
I fail to understand the intension of this code. Could you clarify the intent or simplify the code?